### PR TITLE
[core] Support default aggregate function for partial update and aggregate merge function

### LINKED ISSUE #3373: [Feature] Support default aggregate function for partial update
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Paimon version

Support default aggregate such as last_non_value for partial update with and without sequence group and users can update non-null value with sequence group

### Compute Engine

all

### Minimal reproduce step

Partial update with null value

### What doesn't meet your expectations?

The null value should not be updated in the old record

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by FangYongs**
Fix update null value for partial update with sequence-group

### Purpose

Linked issue: close #3373

### Tests

1. Added test unit PartialUpdateMergeFunctionTest#testSequenceGroupDefaultAggFunc
2. Added test unit PartialUpdateITCase#testSequenceGroupWithDefaultAggFunc and PartialUpdateITCase#testPartialUpdateWithDefaultAndFieldAggregation

### API and Format

no

### Documentation

no

### CODE REVIEW

**Comment by JingsongLi** [comment]
See https://paimon.apache.org/docs/master/primary-key-table/merge-engine/#sequence-group

So we introduce sequence group mechanism for partial-update tables. It can solve:
- Disorder during multi-stream update. Each stream defines its own sequence-groups.
- A true partial-update, not just a non-null update.

True partial-update, means null can update column.

**Comment by FangYongs** [comment]
@JingsongLi As we discussed, I add `fields.default-aggregate-function` for partial update with sequence group, please help to review this PR, thanks cc @liming30

File: paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java [hunk 1]
```diff
@@ -379,6 +380,16 @@ private Map<Integer, FieldAggregator> createFieldAggregators(
                                     isPrimaryKey,
                                     options,
                                     fieldName));
+                } else if (defaultAggFunc != null) {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java:383]: Can you also modify `AggregateMergeFunction`?

Comment (FangYongs) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java:383]: @JingsongLi `AggregatedMergeFunction` is updated too

**Review by liming30 — APPROVED** [review]
+1，after adding this, users can avoid manually configuring `agg` functions one by one.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**